STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.Exceptions.cs

```diff
@@ -0,0 +1,118 @@
+        {
+            try
+            {
+                return await returningHttpExchangeFunction();
+            }
+            catch (NullHttpExchangeException nullHttpExchangeException)
+            {
+                throw CreateHttpExchangeValidationException(
+                    exception: nullHttpExchangeException);
+            }
+            catch (InvalidHttpExchangeRequestException invalidHttpExchangeRequestException)
+            {
+                throw CreateHttpExchangeValidationException(
+                    invalidHttpExchangeRequestException);
+            }
+            catch (InvalidHttpExchangeHeaderException invalidHttpExchangeHeaderException)
+            {
+                throw CreateHttpExchangeValidationException(
+                    invalidHttpExchangeHeaderException);
+            }
+            catch(HttpRequestException httpRequestException)
+            {
+                var failedHttpExchangeException =
+                    new FailedHttpExchangeException(
+                        message: "Failed http request error occurred, contact support.",
+                        innerException: httpRequestException);
+
+                throw CreateHttpExchangeDependencyException(
+                    failedHttpExchangeException);
+            }
+            catch(TaskCanceledException taskCanceledException)
+            {
+                var failedHttpExchangeException =
+                    new FailedHttpExchangeException(
+                        message: "Request timeout error occurred, please contact support.",
+                        innerException: taskCanceledException);
+
+                throw CreateHttpExchangeDependencyException(
+                    failedHttpExchangeException);
+            }
+            catch(ObjectDisposedException objectDisposedException)
+            {
+                var failedHttpExchangeException =
+                    new FailedHttpExchangeException(
+                        message: "Object already disposed error occurred, please fix errors and try again.",
+                        innerException: objectDisposedException);
+
+                throw CreateHttpExchangeDependencyException(
+                    failedHttpExchangeException);
+            }
+            catch(InvalidOperationException invalidOperationException)
+            {
+                var failedHttpExchangeException =
+                    new FailedHttpExchangeException(
+                        message: "Invalid http request operation error occurred, please contact support.",
+                        innerException: invalidOperationException);
+
+                throw CreateHttpExchangeDependencyException(
+                    failedHttpExchangeException);
+            }
+            catch (ArgumentException argumentException)
+            {
+                var failedHttpExchangeException =
+                    new FailedHttpExchangeException(
```

Comment:
Can we map this to local exception `InvalidArgumentHttpExchangeException`?

Reply:
Resolved